packages/js/data/src/setting-options/tests/actions.test.ts

```diff
@@ -0,0 +1,615 @@
+
+		it( 'should handle errors correctly', async () => {
+			const groupId = 'test-group';
+			const settingId = 'test-setting';
+			const error = createTestError( 'API Error' );
+
+			// Mock API error
+			( apiFetch as unknown as jest.Mock ).mockRejectedValue( error );
+
+			registry
+				.dispatch( STORE_NAME )
+				.updateSetting( groupId, settingId, '' );
+
+			await expect(
+				registry
+					.dispatch( STORE_NAME )
+					.saveSetting( groupId, settingId )
+			).rejects.toThrow( error );
+
+			// Verify error state
+			expect( store.getState().errors[ groupId ]?.[ settingId ] ).toBe(
+				error
+			);
+			expect(
+				store.getState().isSaving.settings[ groupId ]?.[ settingId ]
+			).toBe( false );
+		} );
+	} );
+
+	describe( 'saveSettingsGroup', () => {
+		it( 'should handle successful batch update', async () => {
+			const groupId = 'test-group';
+			const mockResults = {
+				update: [
+					createTestSetting( {
+						id: 'setting1',
+						value: 'value1',
+						label: 'Setting 1',
+					} ),
+					createTestSetting( {
+						id: 'setting2',
+						value: 'value2',
+						label: 'Setting 2',
+					} ),
+				],
+			};
+
+			// Mock API response
+			( apiFetch as unknown as jest.Mock ).mockResolvedValue(
+				mockResults
+			);
+
+			registry.dispatch( STORE_NAME ).updateSettings( groupId, [
+				{ id: 'setting1', value: 'value1' },
+				{ id: 'setting2', value: 'value2' },
+			] );
+
+			await registry.dispatch( STORE_NAME ).saveSettingsGroup( groupId );
+
+			expect( apiFetch ).toHaveBeenCalledWith( {
+				path: expect.stringContaining( groupId ),
+				method: 'POST',
+				data: {
+					update: [
+						{ id: 'setting1', value: 'value1' },
+						{ id: 'setting2', value: 'value2' },
+					],
+				},
+			} );
+
+			// Verify state updates
+			expect( store.getState().settings[ groupId ] ).toEqual( {
+				setting1: mockResults.update[ 0 ],
+				setting2: mockResults.update[ 1 ],
+			} );
+			expect( store.getState().isSaving.groups[ groupId ] ).toBe( false );
+		} );
+
+		it( 'should handle partial success in batch update', async () => {
+			const groupId = 'test-group';
+			const mockResults = {
+				update: [
+					createTestSetting( {
+						id: 'setting1',
+						value: 'value1',
+						label: 'Setting 1',
+					} ),
+					{
+						id: 'setting2',
+						error: {
+							code: 'invalid_value',
+							message: 'Invalid value',
+						},
+					},
+				],
+			};
+
+			registry.dispatch( STORE_NAME ).updateSettings( groupId, [
+				{ id: 'setting1', value: 'value1' },
+				{ id: 'setting2', value: 'value2' },
+			] );
+
+			// Mock API response
+			( apiFetch as unknown as jest.Mock ).mockResolvedValue(
+				mockResults
+			);
+
+			await expect(
+				registry.dispatch( STORE_NAME ).saveSettingsGroup( groupId )
+			).rejects.toThrow( 'Failed to update some settings' );
+
+			// Verify successful update
+			expect( store.getState().settings[ groupId ]?.setting1 ).toEqual(
+				mockResults.update[ 0 ]
+			);
+
+			// Verify error state
```

Comment:
In conjunction with the other comment about not caching error values, if that's accepted then we should also verify that the returned value is correct here